src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -212,5 +213,32 @@ public static FormulaValue Guid_UO(IRContext irContext, UntypedObjectValue[] arg
 
             return CommonErrors.RuntimeTypeMismatch(irContext);
         }
+
+        public static async ValueTask<FormulaValue> ForAll_UO(EvalVisitor runner, EvalVisitorContext context, IRContext irContext, FormulaValue[] args)
+        {
+            var arg0 = (UntypedObjectValue)args[0];
+            var arg1 = (LambdaFormulaValue)args[1];
+
+            var itemType = RecordType.Empty().Add(new NamedFormulaType(BuiltinFunction.ColumnName_ValueStr, FormulaType.UntypedObject));
+
+            var resultRows = new List<DValue<RecordValue>>();
+
+            var len = arg0.Impl.GetArrayLength();
+
+            for (var i = 0; i < len; i++)
+            {
```

Comment:
>{ [](http://example.com/codeflow?start=12&length=1)

nit:  should check cancellation token in the loop: 

runner.CheckCancel();

Reply:
Done